### ISSUE TITLE
json output and --transport (0.33)

### ISSUE THREAD

**Issue by aschnell** [doug-gilbert/lsscsi#3]
I'm testing the 0.33 pre-release and am pleased to see the json output. One strange thing: Adding --transport does not add the transport to SCSI devices (only to NVMe devices). I also have to add --list. An oversight?

**Comment by doug-gilbert** [comment]
The only place I see missing transport information is with the scsi_debug driver. Do you have another transport or are you also using scsi_debug to simulate SCSI devices?
To make things a bit more consistent, I have added a new transport called "pseudo_0" just for the scsi_debug driver as its sysfs device is /sys/devices/pseudo_0 (while its bus is "pseudo"). Just pushed that here, it is svn revision 191. Tell me how that goes?
Also interested in feedback about the JSON output as I have no applications that consume storage device JSON. I feel a bit like an author who has written more books than he's read ...

**Comment by aschnell** [comment]
Maybe my description is not so clear. For me the complete 'transport' key is missing in the output
unless I also add --list.

So ```lsscsi --transport --json```:

        {
            "lsscsi_locator": "[0:0:0:0]",
            "host_index": 0,
            "controller_index": 0,
            "target_index": 0,
            "lun": {
                "linux_lun": 0,
                "t10_lun_array": "00 00 00 00 00 00 00 00"
            },
            "peripheral_device_type": {
                "i": 0,
                "meaning": "disk   "
            },
            "primary_device_node": "/dev/sda"
        }

With ```lsscsi --list --transport --json```:

        {
            "lsscsi_locator": "[0:0:0:0]",
            "host_index": 0,
            "controller_index": 0,
            "target_index": 0,
            "lun": {
                "linux_lun": 0,
                "t10_lun_array": "00 00 00 00 00 00 00 00",
                "name": "Logical Unit Number"
            },
            "peripheral_device_type": {
                "i": 0,
                "meaning": "disk   "
            },
            "primary_device_node": "/dev/sda",
            "transport": "sata",
            "wwn": "5002538d007e86ae"
        }

In general the JSON output looks fine. I'm developing libstorage-ng that runs many external programs to gather storage information. Parsing JSON is just so much simpler and more robust that the output for humans.

**Comment by doug-gilbert** [comment]
Okay, I think that I understand your point this time. Thanks for the elaboration. Pushed svn revision 192 and await your response after testing it.
Just in case you are not aware, here are the storage related utilities that now have a JSON output option:
 -  sg_decode_sense, sg_get_lba_status, sg_get_elem_status, sg_inq, sg_logs, sg_luns, sg_opcodes, sg_readcap, sg_rep_zones, sg_ses, sg_vpd [in sg3_utils package]
 - sdparm [sdparm package]
 - lsscsi [lsscsi package]
 - smartctl [smartmontools package]

The latter one (smartctl) is the JSON template (generally speaking) that I have followed. Christian Franke is responsible for the JSON introduction to smartmontools. None of those can take JSON as input. For example sdparm, which handles SCSI Mode page fields, still uses name=value syntax for changing field settings.
